### PR TITLE
build: introduce `alias`, `links`, `upstreamed` fields in schema

### PR DESCRIPTION
**Pull request by nekowinston**
Right now, we have a few ports that are effectively providing ports for other Applications/Websites.

The prime example for that would be `vscode` providing support for both `shiki`, a development library, and `azuredatastudio`. 

This PR would introduce new fields in our ports.yml that allow more freedom in how we point to the port (`alias`, `url`).
Additionally, since some ports are available on official Marketplaces, App Stores, via in-app links, etc., I've added a new `links` object that allows us to track arbitrary URLs, annotated with name and icon.

### CODE REVIEW

**Comment by nekowinston** [comment]
> Just skimming over it right now, do we also want to track what ports have an "upstreamed" version? Maybe like a simple boolean flag? Immediately `WezTerm` and `Zellij` come to mind but I know there are some that I'm missing.

That also came to my mind while working on this. Issues like https://github.com/catppuccin/wezterm/issues/13 make me think that having a doc string would be a good idea. In WezTerm's case, we'd have to tell the user to use the `Catppuccin <Flavor>` version instead of `catppuccin-<flavor>`.

So maybe a `boolean | string`?

File: resources/generate/main.ts [hunk 1]
```diff
@@ -43,19 +64,35 @@ const categorized = Object.entries(ports)
       // create a new array if it doesn't exist
       acc[port.category] ??= [];
 
+      console.log(port);
+
+      let url = port.url;
+      if (!url) {
+        switch (port.type) {
+          case "port": {
+            const repo = port.alias ?? slug;
```
Comment (nekowinston) [inline resources/generate/main.ts:73]: We should check if the alias points to an existing port. JSON schema validation can't do that for us sadly.

**Comment by sgoudham** [comment]
> > Just skimming over it right now, do we also want to track what ports have an "upstreamed" version? Maybe like a simple boolean flag? Immediately `WezTerm` and `Zellij` come to mind but I know there are some that I'm missing.
> 
> That also came to my mind while working on this. Issues like [catppuccin/wezterm#13](https://github.com/catppuccin/wezterm/issues/13) make me think that having an optional doc string would be a good idea. In WezTerm's case, we'd have to tell the user to use the `Catppuccin <Flavor>` version instead of `catppuccin-<flavor>`.
> 
> So maybe a `boolean | string`?

I suppose it depends if the goal of the metadata field here is to 

1. Act purely as a marker to tell the user that the theme is included in the application/tool/whatever

	**OR**

2. Act purely as a marker and give the user installation instructions to account for some of the differences

**Comment by nekowinston** [comment]
Then I think we can definitely go for a boolean now, so we can track what is upstreamed in a single location.

If we decide to add docstring support, we can just extend the schema to `anyOf` `boolean/string` later.

**Pull request merged**